src/vm/wren_vm.c

```diff
@@ -270,26 +275,63 @@ static void closeUpvalue(ObjFiber* fiber)
   fiber->openUpvalues = upvalue->next;
 }
 
-static void bindMethod(WrenVM* vm, int methodType, int symbol,
-                       ObjClass* classObj, Value methodValue)
+// Looks up a foreign method in [moduleName] on [className] with [signature].
+//
+// This will try the host's foreign method binder first. If that fails, it
+// falls back to handling the built-in modules.
+static WrenForeignMethodFn findForeignMethod(WrenVM* vm,
+                                             const char* moduleName,
+                                             const char* className,
+                                             bool isStatic,
+                                             const char* signature)
 {
-  ObjFn* methodFn = IS_FN(methodValue) ? AS_FN(methodValue)
-                                       : AS_CLOSURE(methodValue)->fn;
+  WrenForeignMethodFn fn;
 
-  // Methods are always bound against the class, and not the metaclass, even
-  // for static methods, so that constructors (which are static) get bound like
-  // instance methods.
-  wrenBindMethodCode(classObj, methodFn);
+  // Let the host try to find it first.
+  if (vm->bindForeign != NULL)
+  {
+    fn = vm->bindForeign(vm, moduleName, className, isStatic, signature);
+    if (fn != NULL) return fn;
+  }
 
-  Method method;
-  method.type = METHOD_BLOCK;
-  method.fn.obj = AS_OBJ(methodValue);
+  // Otherwise, try the built-in libraries.
+  #if WREN_USE_LIB_IO
+  fn = wrenBindIOForeignMethod(vm, className, signature);
```

Comment:
This results in a failed assert if the bindForeign function returns NULL and the class is not "IO".